### PR TITLE
FIX-#5767: cast `pathlib.Path` to str for `read_parquet`

### LINKED ISSUE #5767: BUG: Modin fails on reading parquet files with partitioned columns if path specified as a `pathlib.Path` object
### Modin version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest released version of Modin.

- [X] I have confirmed this bug exists on the main branch of Modin. (In order to do this you can follow [this guide](https://modin.readthedocs.io/en/stable/getting_started/installation.html#installing-from-the-github-master-branch).)

### Reproducible Example

```python
import pandas
import modin.pandas as pd

file_name = "modin_issue#0000.parquet"

pandas.DataFrame({"a": [1, 2, 3, 4], "b": [1, 1, 2, 2]}).to_parquet(file_name, partition_cols=["b"])

df = pd.read_parquet(file_name) # 'default to pandas' warning.
print(df)

from pathlib import Path

df = pd.read_parquet(Path(file_name)) # InternalError
print(df)
```

### Issue Description

For some reason, when `Path` is passed modin tries to a read partitioned file anyway and fails as this case is not yet supported.

### Expected Behavior

fallback to pandas on unsupported operation

### Error Logs

<details>

```python-traceback

Traceback (most recent call last):
  File "t3.py", line 14, in <module>
    df = pd.read_parquet(Path(file_name)) # InternalError
  File "repos/modin/modin/logging/logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "repos/modin/modin/pandas/io.py", line 313, in read_parquet
    query_compiler=FactoryDispatcher.read_parquet(
  File "repos/modin/modin/core/execution/dispatching/factories/dispatcher.py", line 180, in read_parquet
    return cls.get_factory()._read_parquet(**kwargs)
  File "repos/modin/modin/core/execution/dispatching/factories/factories.py", line 207, in _read_parquet
    return cls.io_cls.read_parquet(**kwargs)
  File "repos/modin/modin/logging/logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "repos/modin/modin/core/io/file_dispatcher.py", line 164, in read
    if hasattr(query_compiler, "dtypes") and any(
  File "repos/modin/modin/core/storage_formats/pandas/query_compiler.py", line 300, in dtypes
    return self._modin_frame.dtypes
  File "repos/modin/modin/core/dataframe/pandas/dataframe/dataframe.py", line 299, in dtypes
    self._dtypes = self._compute_dtypes()
  File "repos/modin/modin/logging/logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "repos/modin/modin/core/dataframe/pandas/dataframe/dataframe.py", line 318, in _compute_dtypes
    self.tree_reduce(0, lambda df: df.dtypes, dtype_builder)
  File "repos/modin/modin/logging/logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "repos/modin/modin/core/dataframe/pandas/dataframe/dataframe.py", line 126, in run_f_on_minimally_updated_metadata
    result = f(self, *args, **kwargs)
  File "repos/modin/modin/core/dataframe/pandas/dataframe/dataframe.py", line 3343, in to_pandas
    ErrorMessage.catch_bugs_and_request_email(
  File "repos/modin/modin/error_message.py", line 80, in catch_bugs_and_request_email
    raise Exception(
Exception: Internal Error. Please visit https://github.com/modin-project/modin/issues to file an issue with the traceback and the command that caused this error. If you can't file a GitHub issue, please email bug_reports@modin.org.
Internal and external indices on axis 1 do not match.

```

</details>

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5767 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by anmyachev** [comment]
@dchigarev could you review the PR? It's needed for our workflows (even if we have the workarounds).

**Comment by anmyachev** [comment]
@modin-project/modin-core could we merge it?

File: modin/pandas/test/test_io.py [hunk 1]
```diff
@@ -1656,6 +1656,17 @@ def test_read_parquet_2462(self, engine):
...
+    def test_read_parquet_5767(self, engine):
+        test_df = pandas.DataFrame({"a": [1, 2, 3, 4], "b": [1, 1, 2, 2]})
+        file_name = "modin_issue#0000.parquet"
+        with tempfile.TemporaryDirectory() as directory:
+            path = f"{directory}/data"
+            os.makedirs(path)
+            test_df.to_parquet(path + file_name, engine=engine, partition_cols=["b"])
+            read_df = pd.read_parquet(Path(path + file_name))
+        # both Modin and pandas read column "b" as a category
+        df_equals(test_df, read_df.astype("int64"))
```
Comment (YarShev) [inline modin/pandas/test/test_io.py:1668]: Why do we need astype call for read_df?

File: modin/core/io/column_stores/parquet_dispatcher.py [hunk 2]
```diff
@@ -612,6 +613,8 @@ def _read(cls, path, engine, columns, **kwargs):
         ParquetFile API is used. Please refer to the documentation here
         https://arrow.apache.org/docs/python/parquet.html
         """
+        if isinstance(path, pathlib.Path):
```
[the remark this answers is not in the document]
Comment (anmyachev) [reply modin/core/io/column_stores/parquet_dispatcher.py:616]: Pandas also converts to strings, only does it differently.

File: modin/core/io/column_stores/parquet_dispatcher.py [hunk 3]
```diff
@@ -16,6 +16,7 @@
 import os
 import re
 import json
+import pathlib
```
Comment (anmyachev) [inline modin/core/io/column_stores/parquet_dispatcher.py:19]:
```suggestion
```

File: modin/core/io/column_stores/parquet_dispatcher.py [hunk 4]
```diff
@@ -612,6 +613,8 @@ def _read(cls, path, engine, columns, **kwargs):
         ParquetFile API is used. Please refer to the documentation here
         https://arrow.apache.org/docs/python/parquet.html
         """
+        if isinstance(path, pathlib.Path):
+            path = str(path)
```
Comment (anmyachev) [inline modin/core/io/column_stores/parquet_dispatcher.py:617]:
```suggestion
        path = stringify_path(path)
```

File: modin/pandas/test/test_io.py (hunk 1 above)
Comment (YarShev) [reply modin/pandas/test/test_io.py:1668]: ok

**Review by YarShev — APPROVED** [review]

**Comment by anmyachev** [comment]
@YarShev ready to merge

**Review by YarShev — APPROVED** [review]

**Pull request merged**